### ISSUE TITLE
Add support for right to left languages

### ISSUE THREAD

**Issue by nlhkabu** [pypa/warehouse#866]
We need to add support for [languages that are read right to left](https://en.wikipedia.org/wiki/Right-to-left#Current_scripts).

There are a number of solutions that we can investigate for this:

https://www.npmjs.com/package/flipcss
https://github.com/peduarte/rtl-with-sass
https://github.com/MohammadYounes/rtlcss

One thing that we need to bear in mind is that most of the *content* on the site is in English (i.e. the actual package information).  This means that any generated page would be a combination of RTL and LTR languages.

Advice/feedback from the RTL-reading Python community would be appreciated.  What is the best way of combining LTR and RTL content?

**Comment by xavierdutreilh** [comment]
@nlhkabu, the recommended way is to set the `direction` property on your CSS classes (and have a specific class, like `text-rtl` that would change the direction). You can also set the `dir` attribute on any element:
```html
<p>Left-to-right paragraph</p>
<p dir="rtl">Right-to-left paragraph</p>
```  
It would then produce:
<p>Left-to-right paragraph</p>
<p dir="rtl">Right-to-left paragraph</p>

@Bashar, as a user, can you help us on this issue? Do you have any advice on web pages that include LTR and RTL contents? Is there something that we can do to make things simple/easier to read?

**Comment by Bashar** [comment]
@xavierdutreilh in the past we used inline edits line you mentioned but that wasn't scalable due the matter you need sometimes two set of templates.

eventually we reached to two css files that loads different content and each content has its own style="direction:rtl" directive that helps make the content load correctly.

the only challenge was mixed text and numbers in the same line of RTL text in LTR page which will flip phone numbers for example instead being +965 123 4567 it becomes 123 4567 965+ in a RTL section as numbers as you know doesn't get RTL'ed they stay LTR'ed everywhere :)

**Comment by MohammadYounes** [comment]
Since you require having a combination of both directions in the same page, I may suggest https://github.com/rtsao/postcss-rtlcss-combined

As for the mixed text, you can use Character references for directionality and joining control http://www.w3.org/TR/html4/struct/dirlang.html#h-8.2.5 

For example:

```HTML
 <div dir="rtl" > &lrm;+965 123 4567</div>
```

Will render the number as LTR even if inside RTL container.

**Comment by Bashar** [comment]
Thank you @MohammadYounes :+1: just new about &lrm; much appreciated :)

**Comment by dstufft** [comment]
This was supposed to be closed with #1335.
